internal/worker/clienterrors/errors.go

```diff
@@ -58,6 +60,65 @@ func (e *Error) String() string {
 	return fmt.Sprintf("Code: %d, Reason: %s, Details: %v", e.ID, e.Reason, e.Details)
 }
 
+func ensureNoErrs(v reflect.Value) error {
+	switch v.Kind() {
+	case reflect.Interface:
+		if errIf, ok := v.Interface().(error); ok {
+			if errIf != nil {
+				return fmt.Errorf("%v", errIf.Error())
+			}
+		}
+	case reflect.Ptr:
+		if err := ensureNoErrs(v.Elem()); err != nil {
+			return err
+		}
+	case reflect.Struct:
+		for i := 0; i < v.NumField(); i++ {
+			if err := ensureNoErrs(v.Field(i)); err != nil {
+				return err
+			}
+		}
+	case reflect.Slice, reflect.Array:
+		for i := 0; i < v.Len(); i++ {
+			if err := ensureNoErrs(v.Index(i)); err != nil {
+				return err
+			}
+		}
+	case reflect.Map:
+		for _, key := range v.MapKeys() {
+			if err := ensureNoErrs(v.MapIndex(key)); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (e *Error) MarshalJSON() ([]byte, error) {
+	var details interface{}
+	switch v := e.Details.(type) {
+	case error:
+		details = v.Error()
+	case []error:
+		details = fmt.Sprintf("%v", v)
+	default:
+		if err := ensureNoErrs(reflect.ValueOf(v)); err != nil {
+			return nil, fmt.Errorf("found nested error in %+v: %v", e.Details, err)
```

Comment:
This is probably a bit too terse, maybe `found nested error in Details field` or something

Reply:
For the rare cases (none?) where this will happen, I think it's good :-)